### PR TITLE
Account response read time for http prober

### PR DESCRIPTION
**Pull request by bobrik**
Withot this change `transfer` time is near zero, because `trace.end` time is clocked when response body is available, not when it's read.

cc @discordianfish

### CODE REVIEW

File: prober/http.go [hunk 1]
```diff
@@ -296,10 +295,8 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 	if err != nil && resp == nil {
 		level.Error(logger).Log("msg", "Error for HTTP request", "err", err)
 	} else {
-		defer func() {
-			io.Copy(ioutil.Discard, resp.Body)
-			resp.Body.Close()
-		}()
+		requestErrored := err != nil
```
Comment (brian-brazil) [inline prober/http.go]: It'd be usual to put parens around the RHS here

File: prober/http.go [hunk 2]
```diff
@@ -327,6 +324,19 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
...
+			_, err = io.Copy(ioutil.Discard, resp.Body)
+			if err != nil {
+				level.Info(logger).Log("msg", "Failed to read HTTP response body", "err", err)
+				success = false
+			}
+
+			resp.Body.Close()
+		}
+
+		// At this point body is fully read and we can write end time
```
Comment (brian-brazil) [inline prober/http.go]: Full stop

File: prober/http.go [hunk 3]
```diff
@@ -327,6 +324,19 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
...
+		tt.current.end = time.Now()
```
Comment (brian-brazil) [inline prober/http.go]: This is only setting the end time for the last request

Comment (bobrik) [reply prober/http.go]: Considering that only the last request matters for `transfer` times, should I prefix this:

```go
durationGaugeVec.WithLabelValues("transfer").Add(trace.end.Sub(trace.responseStart).Seconds())
```

with this:

```go
if trace.end.IsZero() {
	continue
}
```

or do you have other suggestions?

Comment (brian-brazil) [reply prober/http.go]: That sounds reasonable to me, sound okay to you @discordianfish?

Comment (bobrik) [reply prober/http.go]: I've pushed the change speculatively.

Comment (discordianfish) [reply prober/http.go]: Sounds reasonable

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**